packages/esm-patient-forms-app/src/forms/form-entry.workspace.tsx

```diff
@@ -82,7 +91,21 @@ const FormEntry: React.FC<FormEntryComponentProps> = ({
 
   return (
     <div>
-      {showForm && formInfo && patientUuid && patient && <ExtensionSlot name="form-widget-slot" state={state} />}
+      {showForm && formInfo && patientUuid && patient && htmlForm && encounterUuid && (
+        <HtmlFormEntryWrapper
+          closeWorkspace={closeWorkspace}
+          src={`${window.openmrsBase}/htmlformentryui/htmlform/${htmlForm.formEditUiPage}.page?patientId=${patientUuid}&visitId=${visitUuid}&encounterId=${encounterUuid}&definitionUiResource=${htmlForm.formUiResource}&returnUrl=post-message:close-workspace`}
+        />
+      )}
+      {showForm && formInfo && patientUuid && patient && htmlForm && !encounterUuid && (
+        <HtmlFormEntryWrapper
+          closeWorkspace={closeWorkspace}
+          src={`${window.openmrsBase}/htmlformentryui/htmlform/${htmlForm.formUiPage}.page?patientId=${patientUuid}&visitId=${visitUuid}&definitionUiResource=${htmlForm.formUiResource}&returnUrl=post-message:close-workspace`}
+        />
+      )}
+      {showForm && formInfo && patientUuid && patient && !htmlForm && (
+        <ExtensionSlot name="form-widget-slot" state={state} />
+      )}
```

Comment:
Now, instead of navigating to the O3 app, we render a new HTML Form Entry Wrapper component, which basically just loads the O2 app in an iframe (see below).  

The way I've set up here, by just referencing the Html Form Entry Wrapper within form-entry.workspace, is likely wrong, but I wasn't sure what the best approach was.  Should I simply render a different extension slot ("html-form-entry-widget-slot") and define a new extension thta uses HtmlFormEntryWrapper?  Should there still just be one extension slot ("form-widget-slot") and what gets wired into it varies?  Or should we have an entirely different "html-form-entry.workspace".    Apologies for my naivety with best practices/uses of workspaces. :)

One issue I'm seeing right now, that may factor into this decision, is that if you shift between the form workspace and another workspace (ie the order workspace) the data you can currently entered into the form, but not saved, is *sometimes* lost.  Once we get this set up the "right" way, I can troubleshoot this some more and hopefully we can come up with a workable solution. 

Also note that we pass in a "returnUrl" of "post-message:close-workspace", more on this below.

Reply:
I think the solution to the "losing form data" _may_ be to have a unique workspace for this? Not sure, but I think that's the case.

Reply:
Yeah, let me experiment with doing this in the it's own workspace.